src/core/opamExternalTools.ml

```diff
@@ -0,0 +1,125 @@
+(**************************************************************************)
+(*                                                                        *)
+(*    Copyright 2018 OCamlPro                                             *)
+(*                                                                        *)
+(*  All rights reserved. This file is distributed under the terms of the  *)
+(*  GNU Lesser General Public License version 2.1, with the special       *)
+(*  exception on linking described in the file LICENSE.                   *)
+(*                                                                        *)
+(**************************************************************************)
+
+type t = (string * string list)
+
+let unpack x = x
+let custom x = x
+
+let has_space (cmd, _) = String.contains cmd ' '
+let cmd_to_string (cmd, _) = cmd
+
+let to_string (cmd, args) =
+  (* TODO: Add backslashs to quotes for each args *)
+  let quote x = if String.contains x ' ' then "'" ^ x ^ "'" else x in
```

Comment:
This quoting logic seems like it might be replicated elsewhere in opam. `Bos` also has this

Reply:
Thanks, I'll look around. I tried to be as close as the original code as possible (modulo quoting that is added because that seemed broken anyway)

Reply:
Note that backslash-quoting in single quotes doesn't work for the shell. You have to write e.g. `'foo'\''bar` ; and tabs should also be supported.